src/parser/scanner_support.c

```diff
@@ -88,6 +107,52 @@ struct st_hint_msg
       }
     msg_buf[0] = 0x00;
     is_print = false;
+
+    m_hint_ptr = m_hint_head.next;
+    while (m_hint_ptr)
+      {
+	STMT_HINT *t = m_hint_ptr;
+	m_hint_ptr = t->next;
+	free (t);
+      }
+    memset (&m_hint_head, 0x00, sizeof (STMT_HINT));
+    m_hint_head.stmt_no = -1;
+
+    m_stmt_no = -1;
+    m_select_cnt = -1;
```

Comment:
why m_select_cnt initial value is not 0 ?

Reply:
When used, set it to 0 through the calc_hint_statement_count() function.
It is set to -1 in the initialization state because it can also be identified whether the calc_hint_statement_count() function is called through the value of this variable.